File: src/day25/Day25.kt
package day25

import java.io.File

const val FILENAME = "src/day25/Input.txt"

val input = File(FILENAME).readLines().map { it.toInt() }.toTypedArray()

fun part01(): Long {
    var encryptionKey = 1L
    var value = 1
    while (true) {
        encryptionKey = (encryptionKey * input[1]) % 20201227
        value = (value * 7) % 20201227
        if (value == input[0])
            return encryptionKey
    }
}

fun main() {
    println("Advent of Code 2020, Day 25")
    println(part01())
}

File: day25/day25.go
package day25

import (
	"advent-of-code-2020/utils"
)

const filename = "./inputs/day25.txt"

var input = utils.MapToInt(utils.ReadLines(filename))

func part01() int {
	encryptionKey := 1
	value := 1
	for {
		encryptionKey = (encryptionKey * input[1]) % 20201227
		value = (value * 7) % 20201227
		if value == input[0] {
			return encryptionKey
		}
	}
}

func Main() (int, func() int, func() int) {
	return 25, part01, func() int {
		// no part02 on day25
		return -1
	}
}
